corrected minor grammar mistakes in README.md

ConnorMooneyhan opened the pull request:
<!--
If your PR closes some issues, please write `Closes #XXXX`
where `XXXX` is the number of the issue you want to fix.
Each issue goes on its own line.
-->

pksunkara reviewed (commented):
I am not actually sure of some of these since English is not my native language. Can you point me to some sources?

ConnorMooneyhan commented:
> I am not actually sure of some of these since English is not my native language. Can you point me to some sources?

Sure thing.

I changed "applications" to "application's" since we're using the possessive form of "application" here, which needs an apostrophe. ( https://www.ef.edu/english-resources/english-grammar/forming-possessive/ )

I changed "user provided" to "user-provided" and "argument parsing" to "argument-parsing" because these are compound adjectives. ( https://www.grammarly.com/blog/compound-adjectives/ )

I changed "many multiple levels of subcommands" to just "multiple levels of subcommands" because "many" and "multiple" mean about the same thing and "multiple" is more natural here because there's a countable number of things it's describing. ( https://diffsense.com/diff/many/multiple )

I changed "if you want a custom help, version, or usage statements" to "if you want custom help, version, or usage statements" because "statements" is plural so "a" is incorrect here. The other option would be to change it to "if you want a custom help, version, or usage **statement**", where statement is plural to match the "a". I preferred "statements" being plural because there are many possible statements the user could use this to make.

I hope this helps!
Connor

pksunkara reviewed (approved):
bors r+
